app/src/main/java/com/example/blah/mobilestudio/Activities/MainActivity.java

```diff
@@ -29,28 +35,65 @@
 
 public class MainActivity extends AppCompatActivity implements FolderStructureFragment.OnFileSelectedListener, OnItemSelectedListener {
 
-    // Value for the fragments to call when they need the root folder string
-    public static final String ROOT_FOLDER = "root folder";
-    // Value for fragments to call when they need the current folder string
-    public static final String CURRENT_DIRECTORY = "current directory";
-
+    // Constants to pass data to other fragments and activities
     public static final int PICK_DIRECTORY_REQUEST = 1;
     public static final int WRITE_EXTERNAL_STORAGE_PERMISSION_REQUEST = 2;
+    // The smallest/largest amount of pixels that a view can be resized to.
+    public static final float REGION_OFFSET = 100f;
 
-    private String currentFolder;
     private String rootFolder;
 
-    // The various UI elements available
     Toolbar toolbar;
+
+    // Fragments
     FolderStructureFragment folderStructureFragment;
     FileFragment fileFragment;
+    private ResizerFragment horizontalResizerFragment;;
+    private ResizerFragment veritcalResizerFragment;
+    private AndroidMonitorFragment androidMonitorFragment;
     private BreadcrumbFragment breadFragment;
 
+    // The orientation of the screen
+    private boolean isLandscape;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        setUpUI();
+
+        if (rootFolder == null) {
+            rootFolder = "";
+        }
+
+        handlePermissions();
+    }
+
+    // Sets up the UI of the Main Activity
+    // Initialises the toolbar, the orientation and all of the fragments.
+    private void setUpUI(){
+        initialiseOrientation();
+        initialiseToolbar();
+
+        setResizers();
+
+        initialiseFragments();
+
```

Comment:
There is some extra spaces here!, this file needs some reformatting the code and adjusting the line spaces.